File: ex_003.py
# Сложность O(1)

def sum(a, b):
    return a + b

print(sum(1, 1))

# Сложность O(n) - создание массива

def create_array(n):
    arr = []
    
    for i in range(n):
        arr.append(i)
    
    return arr 

print(create_array(5))

# Рекурсия - тоже сложность O(n)

# Рекурсия: функция вызывает саму себя

def recursiveFunction(n):
    print(n)
    if n == 0:
        return
    
    recursiveFunction(n-1)
    
print(recursiveFunction(5))

File: ex_003.js
// Пример пространственный сложности O(1)

function sum(a, b) {
  return a + b;
}

// Примеры линейной сложности

function createArray(n) {
  let arr = [];
  for (let i = 0; i < n; i++) {
    arr.push(i);
  }
  return arr;
}

// Случай рекурсии - тоже линейная сложность

function recursiveFunction(n) {
  console.log(n);
  if (n === 0) return;
  recursiveFunction(n - 1);
}

console.log(sum(1, 1));

console.log(createArray(5));

console.log(recursiveFunction(5));
